src/core/nestedObjectToArray.js

```diff
@@ -0,0 +1,25 @@
+export default function nestedObjectToArray(nestedObject) {
+  const nestedObjectArray = [];
+
+  function recuresiveNestedObjectToArray(nestedObject, depth = 1) {
+    nestedObject.forEach((document) => {
+      if (document.documents.length !== 0) {
+        nestedObjectArray.push({
+          id: document.id,
+          title: document.title,
+          depth: depth,
+        });
+        recuresiveNestedObjectToArray(document.documents, depth + 1);
+      } else {
+        nestedObjectArray.push({
+          id: document.id,
+          title: document.title,
+          depth: depth,
+        });
+      }
```

Comment:
```suggestion
      nestedObjectArray.push({
        id: document.id,
        title: document.title,
        depth: depth,
      });

      if (document.documents.length !== 0) {
        recuresiveNestedObjectToArray(document.documents, depth + 1);
      }
```

이렇게 작성해도 괜찮을 것 같아요..! 🤔